src/modules/mc_att_control/AttitudeControl/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+############################################################################
+#
+#   Copyright (c) 2019 PX4 Development Team. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in
+#    the documentation and/or other materials provided with the
+#    distribution.
+# 3. Neither the name PX4 nor the names of its contributors may be
+#    used to endorse or promote products derived from this software
+#    without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+# OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+# AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+############################################################################
+
+px4_add_library(AttitudeControl
+	AttitudeControl.cpp
+)
+target_include_directories(AttitudeControl
+PUBLIC
+	${CMAKE_CURRENT_SOURCE_DIR}
```

Comment:
Actually, is this right? Shouldn't it be `INTERFACE`?

Reply:
I'll look it up, thanks.

Reply:
Today I learned from [stackoverflow](https://stackoverflow.com/questions/26243169/cmake-target-include-directories-meaning-of-scope) (the [cmake doc](https://cmake.org/cmake/help/v3.0/command/target_include_directories.html) is not as clear):
`PRIVATE`: only include for compiling the target
`INTERFACE`: only include for compiling dependencies
`PUBLIC`: include in both cases
With `INTERFACE` it fails to include the header in `AttitudeControl.cpp` with `#include <AttitudeControl.hpp>` but works with the explicit path `#include "AttitudeControl.hpp"`. I thought the first method combined with `PUBLIC` is preferable for the case you change the file structure of the target library, does anything speak against it?

Reply:
After a bit more research I consider this assumption to be confirmed.

Reply:
Makes sense. In other parts of the code base this hasn't come up because we have the top level `src` and `src/lib` included. The way you've done it is better and gives us more control.